src/decimal.rs

```diff
@@ -664,6 +624,45 @@ impl ToPrimitive for Decimal {
     }
 }
 
+impl fmt::Display for Decimal {
+    fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+        // Get the scale - where we need to put the decimal point
+        let scale = self.scale();
+
+        // Get the whole number without decimal points (or signs)
+        let uint = self.to_biguint();
+
+        // Convert to a string and manipulate that (neg at front, inject decimal)
+        let mut rep = uint.to_string();
+        let len = rep.len();
+
+        // Inject the decimal point
+        if scale > 0 {
+            // Must be a low fractional
+            if scale > len as u32 {
+                let mut new_rep = String::new();
+                let zeros = repeat("0").take(scale as usize - len).collect::<String>();
+                new_rep.push_str("0.");
+                new_rep.push_str(&zeros[..]);
+                new_rep.push_str(&rep[..]);
+                rep = new_rep;
+            } else if scale == len as u32 {
+                rep.insert(0, '.');
+                rep.insert(0, '0');
+            } else {
+                rep.insert(len - scale as usize, '.');
+            }
+        }
+
+        // Last set the negation
+        if self.is_negative() {
+            rep.insert(0, '-');
+        }
+
+        f.pad(&rep)
```

Comment:
If I'm not mistaken, this `pad` follows certain format directives unlike `write_str` which simply writes the string.  For example, `format!("{:<30}", d)` should right-align the decimal number in a column that is 30 characters wide and `write_str` would not do that.